### PR TITLE
[#80] 009-quarkus-infinispan-grpc-kafka port

### PR DESCRIPTION
**Pull request by mjurc**
* Migrating 009-quarkus-infinispan-grpc-kafka from beefy-scenarios to messaging/infinispan-grpc-kafka.

### CODE REVIEW

File: messaging/infinispan-grpc-kafka/pom.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.quarkus.ts.qe</groupId>
+        <artifactId>parent</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+        <relativePath>../..</relativePath>
+    </parent>
+    <artifactId>infinispan-grpc-kafka</artifactId>
```
Comment (Sgitario) [inline messaging/infinispan-grpc-kafka/pom.xml:10]:
```suggestion
    <artifactId>messaging-infinispan-grpc-kafka</artifactId>
```

File: messaging/infinispan-grpc-kafka/pom.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    <name>Quarkus QE TS: Infinispan, GRPC, Kafka</name>
```
Comment (Sgitario) [inline messaging/infinispan-grpc-kafka/pom.xml:11]:
```suggestion
    <name>Quarkus QE TS: Messaging: Infinispan, GRPC, Kafka</name>
```

File: messaging/infinispan-grpc-kafka/src/test/java/io/quarkus/qe/books/BookServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.qe.containers.InfinispanTestResource;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.http.ContentType;
+
+// TODO: change to @QuarkusScenario once https://github.com/quarkus-qe/quarkus-test-framework/issues/292 is solved
```
Comment (Sgitario) [inline messaging/infinispan-grpc-kafka/src/test/java/io/quarkus/qe/books/BookServiceTest.java:15]: I don't think this issue should block using QuarkusScenario here. You can use `@Container` plus `::resource` for external resources to be included in the container instance.

**Review by Sgitario — CHANGES_REQUESTED** [review]
Also, please use an unique package name like `io.quarkus.qe.messaging.infinispan` for all the sources.

**Comment by Sgitario** [comment]
As this module needs Kafka SSL and SASL, we can't use Quarkus scenario as it's unsupported: https://github.com/quarkus-qe/quarkus-test-framework/issues/248

For the record, I could configure the scenario to only use Infinispan:

```
@Container(image = "infinispan/server:11.0.4.Final-2", expectedLog = "Infinispan Server.*started in", port = 11222)
static DefaultService infinispan = new DefaultService()
        .withProperty("IDENTITIES_PATH", "resource::/identities.yaml")
        .withProperty("CONFIG_PATH", "resource::/config.yaml")
        .withProperty("SERVER_JKS_PATH", "resource::/server.jks");

@QuarkusApplication
static RestService app = new RestService()
        .withProperty("quarkus.infinispan-client.server-list",
                () -> infinispan.getHost().replace("http://", "") + ":" + infinispan.getPort());
```

File: messaging/infinispan-grpc-kafka/pom.xml (hunk 1 above)
Comment (mjurc) [reply messaging/infinispan-grpc-kafka/pom.xml]: Fixed!

File: messaging/infinispan-grpc-kafka/pom.xml (hunk 2 above)
Comment (mjurc) [reply messaging/infinispan-grpc-kafka/pom.xml]: Fixed!

**Comment by mjurc** [comment]
> Also, please use an unique package name like `io.quarkus.qe.messaging.infinispan` for all the sources.

Updated.

**Review by Sgitario — APPROVED** [review]

**Pull request merged**